core/src/main/scala/akka/kafka/internal/ConsumerStage.scala

```diff
@@ -419,22 +418,27 @@ private[kafka] abstract class ConsumerStageLogic[K, V, Out](
         None
     }
     // close() is blocking (stupid api that doesn't take timeout parameters)
-    // WakeupException will be thrown by close() if the wakeup from stopTimeout is triggered
+    // WakeupException will be thrown by close() if the wakeup from shutdownTimeout is triggered
     Try(consumer.close())
-    stopTimeoutTask.foreach(_.cancel())
-    stoppedPromise.success(Done)
+    shutdownTimeoutTask.foreach(_.cancel())
+    shutdownPromise.trySuccess(Done)
     super.postStop()
```

Comment:
@13h3r perhaps you missed my previous comment. There is a need for `stopPromise.trySuccess(Done)` here also.